### PR TITLE
Fix bug #1355

### PR DESCRIPTION
**Pull request by abhisrkckl**
Fixes issue #1355

### CODE REVIEW

File: src/pint/pint_matrix.py [hunk 1]
```diff
@@ -607,8 +607,7 @@ def combine_design_matrices_by_param(matrix1, matrix2, padding=0.0):
     for d_param in matrix2.derivative_params:
         if d_param in base_params:
             raise ValueError(
-                "Input design matrix {} has duplicated "
-                "parameters with matrix {}".format(ii, 0)
+                "Unable to combine the two design matrices as they have duplicated parameters."
```
Comment (aarchiba) [inline src/pint/pint_matrix.py:610]: Please ensure that the new message informs the user which parameters are duplicated like the old one did.

File: tests/test_combine_design_matrices_by_param.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+@pytest.fixture()
+def model_and_toas():
+    parfile = examplefile("B1855+09_NANOGrav_dfg+12_TAI.par")
+    timfile = examplefile("B1855+09_NANOGrav_dfg+12.tim")
+    model, toas = get_model_and_toas(parfile, timfile)
+    return model, toas
+
+
+def test_combine_design_matrices_by_param(model_and_toas):
```
Comment (aarchiba) [inline tests/test_combine_design_matrices_by_param.py:18]: What property of the function does this test verify? If it's verifying more than one, please split it into separate tests so we can tell where failures occur.

File: src/pint/pint_matrix.py (hunk 1 above)
Comment (abhisrkckl) [reply src/pint/pint_matrix.py]: I have changed this to mention `d_param` in the error message.

File: tests/test_combine_design_matrices_by_param.py (hunk 2 above)
Comment (abhisrkckl) [reply tests/test_combine_design_matrices_by_param.py:18]: I have renamed this test and added a docstring to explain what is being tested.

File: src/pint/pint_matrix.py [hunk 3]
```diff
@@ -607,8 +607,7 @@ def combine_design_matrices_by_param(matrix1, matrix2, padding=0.0):
     for d_param in matrix2.derivative_params:
         if d_param in base_params:
             raise ValueError(
-                "Input design matrix {} has duplicated "
-                "parameters with matrix {}".format(ii, 0)
+                f"Unable to combine the two design matrices as they have duplicated parameter(s). The duplicated parameter is {d_param}."
```
Comment (dlakaplan) [inline src/pint/pint_matrix.py:610]: Could I suggest a slight modification to make it easier to parse:

f"Unable to combine the two design matrices as they have duplicated parameter(s): '{d_param}'"

Comment (abhisrkckl) [reply src/pint/pint_matrix.py]: Done.

**Comment by dlakaplan** [comment]
Also, can you add something to `CHANGELOG.md`?

**Pull request merged**